src/stream/src/executor/watermark_filter.rs

```diff
@@ -220,15 +228,16 @@ impl<S: StateStore> WatermarkFilterExecutor<S> {
 
                         // Take the global max watermark when scaling happens.
                         if previous_vnode_bitmap != vnode_bitmap {
-                            current_watermark =
-                                Self::get_global_max_watermark(&table, &global_watermark_table)
-                                    .await?;
+                            current_watermark = Self::get_global_max_watermark(
```

Comment:
We may reach deadlock here if we do read committed on epoch.prev before yielding the barrier, because committed this prev epoch depends on collecting this barrier.